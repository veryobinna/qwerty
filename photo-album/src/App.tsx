import React, { Component } from 'react';
import Header from './components/Header';
import Footer from './components/Footer';
import Photos from './components/Photos';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Thumbnails from './components/Thumbnails';
import NotFound from './components/NotFound';
import ErrorBoundary from './ErrorBoundary';
import { PHOTO_API } from './constants';
class AppRoutes extends Component {
  state = {
    photos: [],
  };

  componentDidMount() {
    fetch(PHOTO_API)
      .then((resp) => resp.json())
      .then((resp) => this.setState({ photos: resp }))
      .catch((err) => console.error(err));
  }

  render(): React.ReactNode {
    return (
      <Routes>
        <Route
          path="/"
          element={<Thumbnails thumbnails={this.state.photos} />}
        />
        <Route
          path="/photos/:id"
          element={<Photos photos={this.state.photos} />}
        />
        <Route path="*" element={<NotFound />} />
      </Routes>
    );
  }
}

export default function App() {
  return (
    <Router>
      <div className="flex flex-col h-screen justify-between">
        <div>
          <Header />
          <ErrorBoundary>
            <AppRoutes />
          </ErrorBoundary>
        </div>
        <Footer />
      </div>
    </Router>
  );
}
