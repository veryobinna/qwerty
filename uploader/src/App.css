*{
  box-sizing: border-box;
}
.header{
  text-align: center;
  font-size: 30px;
  font-weight: bold;
  color: #303030;
  border-bottom: #e5e3e3 1px solid;
  padding: 10px;
}
.upload-input{
  border: 1px solid #e9e7e7;
  padding: 20px;
  font-size: 16px;
  border-radius: 10px;
  background-color: #c9c9c9;
  width: 100%;
}
.relative{
  position: relative;
  width: 80%;
}
.upload-container{
  display: flex;
  justify-content: center;
  height: 30rem;
  align-items: center;
  flex-direction: column;

}
.upload-button{
  padding: 10px;
  background-color: #5317d1;
  width: 7rem;
  font-weight: bold;
  color: #eaebef;
  border: 1px solid #a19e9e;
  border-radius: 5px;
  letter-spacing: 1px;
  margin: 5px;
}
.upload-button:hover{
  background-color: #5c1de1;
  cursor: pointer;
  color: #fbfbfb;
  border: 1px solid #c9c8c8;;
}
.error{
  color: red;
  width: 80%;
  font-size: small;
}
@media screen and (min-width: 820px) {
  .relative{
    position: relative;
    width: 40%;
  }
  .error{
    width: 40%;
  }
  .upload-button{
    position: absolute;
    top: 14px;
    right: 20px;
    margin: 0;

  }
}