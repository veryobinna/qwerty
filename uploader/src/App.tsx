import React from 'react';

import './App.css';
import { Header } from './components/Header';
import Uploader from './components/Uploader';

function App() {
  return (
    <div>
      <Header />
      <Uploader />
    </div>
  );
}

export default App;
