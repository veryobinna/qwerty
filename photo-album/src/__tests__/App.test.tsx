import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import App from '../App';

test('renders app', () => {
  const r = createRenderer();
  r.render(<App />);
  fetchMock.mockOnce(JSON.stringify([]));
  expect(r.getRenderOutput()).toMatchSnapshot();
});

