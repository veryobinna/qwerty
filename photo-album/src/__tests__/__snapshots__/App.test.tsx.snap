// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
<BrowserRouter>
  <div
    className="flex flex-col h-screen justify-between"
  >
    <div>
      <Header />
      <ErrorBoundary>
        <AppRoutes />
      </ErrorBoundary>
    </div>
    <Footer />
  </div>
</BrowserRouter>
`;
